### ISSUE TITLE
Random seed for reproducible RandomLinkSplit

### ISSUE THREAD

**Issue by davidfstein** [pyg-team/pytorch_geometric#6820]
### 🚀 The feature, motivation and pitch

It would be helpful to have an option to pass a seed to the RNG to allow for reproducible splits with the RandomLinkSplit function. 

It seems that ```torch.randperm``` is the source of randomness. Could be adjusted as
```
g = torch.Generator(device=device)
g.manual_seed(SEED)
torch.randperm(..., generator=g)
```
and then allow the user to pass the seed as an argument. I wasn't sure if there was a source of randomness for the negative sampling but that would need to be addressed as well if so.

**Comment by rusty1s** [comment]
Yeah, ideally we would need to extend all functions that utilize random sampling via a `generator` method. This would be a more of a long-term project so a quick fix would be to just specify the seed globally?

**Comment by mgao97** [comment]
Got it! Thank you!

**Comment by davidfstein** [comment]
Ah I didn't realize you could set the torch seed globally, thats good enough for what I need to do
